### PR TITLE
Add migration for indexing params column and adding NOT NULL constraint

### PR DESCRIPTION
**Pull request by pinin4fjords**
This PR alters the generic scxa_coords table to require json parameterisation. I'm in the process of populating this field from the test field (test and prod), should be done soon. 

I've also added and index on the JSONB column to allow us to e.g. simply check for the presence of certain top-level keys (e.g. perplexity). 

EDIT: I've also corrected the CI test script, which wasn't dying on failure.

### CODE REVIEW

**Comment by pinin4fjords** [comment]
For posterity, here's how I'm populating the field:

```
echo "SELECT distinct parameterisation FROM scxa_coords" | psql -At $dbConnection | sort | while read -r l; do
    param=$(echo "$l" | awk -F'=' '{print $1}'); 
    val=$(echo "$l" | awk -F'=' '{print $2}'); 
    json="[{\"${param}\": $val}]"; 
    echo "$json"; echo "UPDATE scxa_coords SET json_parameterisation = '$json' WHERE parameterisation='$l'" | psql $dbConnection; 
done
```
(there's probably some postgres formatting logic that could have done this quicker)

**Comment by pinin4fjords** [comment]
Hmm, something wrong actually- test failing even though we have the green tick

**Comment by pcm32** [comment]
This looks good to me. I wonder if you want to complete the loading code against this branch of schema, and then we merge when you are happy with both? I could imagine that when trying to load you might want to make some changes here.

**Comment by pinin4fjords** [comment]
> This looks good to me. I wonder if you want to complete the loading code against this branch of schema, and then we merge when you are happy with both? I could imagine that when trying to load you might want to make some changes here.

Good plan!

**Comment by pcm32** [comment]
I thought that you had applied these changes though to test and wwwdev, which I'm not sure if they are up-to-date now with what it is here? Maybe that will require some back and forth with flyway?

**Comment by pcm32** [comment]
ahh, no, you pushed the previous step to wwwdev and test... nevermind me.

File: flyway/scxa/migrations/V15__scxa_add_coords_param_index_not_null.sql [hunk 1]
```diff
@@ -0,0 +1,4 @@
+ALTER TABLE scxa_coords ALTER COLUMN json_parameterisation SET NOT NULL;
```
Comment (alfonsomunozpomer) [inline flyway/scxa/migrations/V15__scxa_add_coords_param_index_not_null.sql:1]: I’m not a fan of Hungarian notation. Maybe we could just use the same `parameterisation` field name if we’re not going to keep the old column?

**Review by alfonsomunozpomer — APPROVED** [review]
I’m not a fan of Hungarian notation. Maybe we could just use the same parameterisation field name if we’re not going to keep the old column? Your call, though.

File: flyway/scxa/migrations/V15__scxa_add_coords_param_index_not_null.sql [hunk 2]
```diff
@@ -0,0 +1,4 @@
...
+ALTER TABLE scxa_coords DROP COLUMN parameterisation;
```
Comment (pcm32) [inline flyway/scxa/migrations/V15__scxa_add_coords_param_index_not_null.sql:2]:
```suggestion
ALTER TABLE scxa_coords DROP COLUMN parameterisation;
ALTER TABLE scxa_coords ALTER COLUMN json_parameterisation SET NOT NULL;
ALTER TABLE scxa_coords RENAME COLUMN json_parameterisation TO  parameterisation;
```

File: flyway/scxa/migrations/V15__scxa_add_coords_param_index_not_null.sql [hunk 3]
```diff
@@ -0,0 +1,4 @@
...
+
+CREATE INDEX scxa_coords_json_parameterisation ON scxa_coords using gin(json_parameterisation);
```
Comment (pcm32) [inline flyway/scxa/migrations/V15__scxa_add_coords_param_index_not_null.sql:4]:
```suggestion
CREATE INDEX scxa_coords_parameterisation ON scxa_coords using gin(parameterisation);
```

**Comment by pinin4fjords** [comment]
See paired changes in loading code https://github.com/ebi-gene-expression-group/db-scxa/pull/51

**Comment by pinin4fjords** [comment]
> This looks good to me. I wonder if you want to complete the loading code against this branch of schema, and then we merge when you are happy with both? I could imagine that when trying to load you might want to make some changes here.

Think we're good to go @pcm32

**Review by pcm32 — APPROVED** [review]
Looks good!

**Pull request merged**